sotodlib/io/bookbinder.py

```diff
@@ -340,14 +346,25 @@ def preprocess(self):
 
             t = get_frame_times(frame)[1]
             ts.append(t)
+            smurf_frame_counters.append(frame['primary'].data[fc_idx])
             frame_idxs.append(np.full(len(t), frame_idx, dtype=np.int32))
 
             self.nframes += 1
             frame_idx += 1
 
         self.times = np.hstack(ts)
+        self.smurf_frame_counters = np.hstack(smurf_frame_counters)
         self.frame_idxs = np.hstack(frame_idxs)
 
+        # If low-precision, we need to linearize timestamps in order for
+        # bookbinder to work properly
+        if self.timing_paradigm == 'Low Precision':
+            self.log.info(
+                "Timestamps are Low Precision, linearizing from frame-counter"
+            )
+            fsamp, offset = np.polyfit(self.smurf_frame_counters, self.times, 1)
```

Comment:
Maybe not `fsamp` ... it's sample_period, or step, or dt or something.

Reply:
done